### PR TITLE
MEQP1 fixes: Replacing some discouraged functions: curl_exec(), file_exists(), mkdir(), unlink()

### LINKED ISSUE #1055: Apply MEQP1 code standar

### PR DESCRIPTION
**Pull request by roberto-ebizmarts**
Using Varien_Http_Adapter_Curl methods, instead using discouraged curl_exec function. closes #1055

### CODE REVIEW

**Comment by roberto-ebizmarts** [comment]
ref. #779

File: app/design/frontend/base/default/template/ebizmarts/mailchimp/popup/emailcatcher.phtml [hunk 1]
```diff
@@ -219,7 +219,9 @@
         }
         <?php endif; ?>
         function validateEmail(email) {
-            var re = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+            var user = '([^<>()[\\]\\\\.,;:\\s@\\"]+(\\.[^<>()[\\]\\\\.,;:\\s@\\"]+)*)|(\\".+\\")';
+            var domain = '(\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,})';
+            var re = new RegExp('ˆ' + user + '@' + domain + '$');
```
Comment (roberto-ebizmarts) [inline app/design/frontend/base/default/template/ebizmarts/mailchimp/popup/emailcatcher.phtml]: Here was solved:
`Line exceeds 120 characters in Reg-Exp (emailcatcher.phtml).`
 ref. #779

File: app/code/community/Ebizmarts/MailChimp/Helper/Data.php [hunk 2]
```diff
@@ -3024,8 +3081,9 @@ public function getScopeByMailChimpStoreId($mailChimpStoreId)
         $collection = Mage::getResourceModel('core/config_data_collection')
             ->addFieldToFilter('path', array('eq' => Ebizmarts_MailChimp_Model_Config::GENERAL_MCSTOREID))
             ->addFieldToFilter('value', array('eq' => $mailChimpStoreId));
+
         if ($collection->getSize()) {
-            $configEntry = $collection->getFirstItem();
+            $configEntry = $collection->setPageSize(1, 1)->getLastItem();
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/Helper/Data.php]: Move setPageSize filter to the collection definition to keep everything together.
setPageSize function only takes 1 parameter.

File: app/code/community/Ebizmarts/MailChimp/Helper/Data.php [hunk 3]
```diff
@@ -3045,8 +3103,9 @@ public function getFirstScopeFromConfig($path, $value)
         $collection = Mage::getResourceModel('core/config_data_collection')
             ->addFieldToFilter('path', array('eq' => $path))
             ->addFieldToFilter('value', array('eq' => $value));
+
         if ($collection->getSize()) {
-            $configEntry = $collection->getFirstItem();
+            $configEntry = $collection->setPageSize(1, 1)->getLastItem();
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/Helper/Data.php]: Move setPageSize filter to the collection definition to keep everything together.
setPageSize function only takes 1 parameter.

File: app/code/community/Ebizmarts/MailChimp/Helper/Data.php [hunk 4]
```diff
@@ -5024,7 +5122,18 @@ public function ping($storeId)
...
+        $itemType,
+        $syncDelta,
+        $syncError,
+        $syncModified,
+        $syncDeleted,
+        $token,
+        $syncedFlag,
+        $deletedRelatedId,
+        $allowBatchRemoval,
+        Varien_Object $ecommerceSyncDataItem
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/Helper/Data.php]: There is a specific class for this.

File: app/code/community/Ebizmarts/MailChimp/Model/Api/Subscribers/MailchimpTags.php [hunk 5]
```diff
@@ -588,11 +588,13 @@ protected function addCompany($customAtt, $customer, $key)
...
     {
         $lastOrder = $this->getLastOrder();
+
         if ($lastOrder === null) {
             $helper = $this->getMailchimpHelper();
             $orderCollection = $helper->getOrderCollectionByCustomerEmail($this->getSubscriber()->getSubscriberEmail());
+
             if ($this->isNotEmptyOrderCollection($orderCollection)) {
-                $lastOrder = $orderCollection->setOrder('created_at', 'DESC')->getFirstItem();
+                $lastOrder = $orderCollection->setOrder('created_at', 'DESC')->setPageSize(1, 1)->getLastItem();
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/Model/Api/Subscribers/MailchimpTags.php]: Move setPageSize filter to the collection definition to keep everything together.
setPageSize function only takes 1 parameter.

File: app/code/community/Ebizmarts/MailChimp/etc/system.xml [hunk 6]
```diff
@@ -37,7 +37,8 @@
                             <label>Enabled</label>
                             <frontend_type>select</frontend_type>
                             <source_model>adminhtml/system_config_source_yesno</source_model>
-                            <comment><![CDATA[By enabling this extension, you agree to <a href="https://mailchimp.com/legal/additional-terms/" target="_blank">Mailchimp's additional terms</a>.]]></comment>
+                            <comment>
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/etc/system.xml]: This blank lines after the start tag cause failure.

File: app/code/community/Ebizmarts/MailChimp/etc/system.xml [hunk 7]
```diff
@@ -112,7 +113,8 @@
                             <show_in_website>1</show_in_website>
                             <show_in_store>1</show_in_store>
                             <can_be_empty>1</can_be_empty>
-                            <comment><![CDATA[Synchronize Magento's General Subscription Audience with this Mailchimp audience.]]></comment>
+                            <comment>
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/etc/system.xml]: This blank lines after the start tag cause failure.

File: app/code/community/Ebizmarts/MailChimp/etc/system.xml [hunk 8]
```diff
@@ -133,7 +135,8 @@
                             <show_in_default>1</show_in_default>
                             <show_in_website>1</show_in_website>
                             <show_in_store>1</show_in_store>
-                            <comment><![CDATA[In case no Interest groups are selected it will work as a normal subscription to the configured audience]]></comment>
+                            <comment>
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/etc/system.xml]: This blank lines after the start tag cause failure.

File: app/code/community/Ebizmarts/MailChimp/etc/system.xml [hunk 9]
```diff
@@ -187,7 +190,8 @@
                             <show_in_default>1</show_in_default>
                             <show_in_website>1</show_in_website>
                             <show_in_store>1</show_in_store>
-                            <comment><![CDATA[Show Newsletter Subscribe checkbox in the last Checkout Step (Order Review).]]></comment>
+                            <comment>
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/etc/system.xml]: This blank lines after the start tag cause failure.

File: app/code/community/Ebizmarts/MailChimp/etc/system.xml [hunk 10]
```diff
@@ -197,7 +201,8 @@
                             <show_in_default>1</show_in_default>
                             <show_in_website>1</show_in_website>
                             <show_in_store>1</show_in_store>
-                            <comment><![CDATA[This is required in order to tell the extensions what customer attributes should be sent to your selected Mailchimp list. <a class="mc-link" target="_blank" href="http://wiki.ebizmarts.com/m4m-merge-fields">See recommended configuration here.</a>]]></comment>
+                            <comment>
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/etc/system.xml]: This blank lines after the start tag cause failure.

File: app/code/community/Ebizmarts/MailChimp/etc/system.xml [hunk 11]
```diff
@@ -207,7 +212,8 @@
                             <show_in_default>1</show_in_default>
                             <show_in_website>1</show_in_website>
                             <show_in_store>1</show_in_store>
-                            <comment><![CDATA[Click this button to create the above fields on the selected Mailchimp audience.]]></comment>
+                            <comment>
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/etc/system.xml]: This blank lines after the start tag cause failure.

File: app/code/community/Ebizmarts/MailChimp/etc/system.xml [hunk 12]
```diff
@@ -217,7 +223,8 @@
                             <show_in_default>1</show_in_default>
                             <show_in_website>1</show_in_website>
                             <show_in_store>1</show_in_store>
-                            <comment><![CDATA[Click this button to create the webhook manually, it will only work if the webhook doesn't already exist.]]></comment>
+                            <comment>
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/etc/system.xml]: This blank lines after the start tag cause failure.

File: app/code/community/Ebizmarts/MailChimp/etc/system.xml [hunk 13]
```diff
@@ -238,7 +245,8 @@
                             <show_in_website>1</show_in_website>
                             <show_in_store>1</show_in_store>
                             <can_be_empty>1</can_be_empty>
-                            <comment><![CDATA[If this is set to "No", the synchronization will be from Magento's newsletter to Mailchimp only.]]></comment>
+                            <comment>
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/etc/system.xml]: This blank lines after the start tag cause failure.

File: app/code/community/Ebizmarts/MailChimp/etc/system.xml [hunk 14]
```diff
@@ -249,7 +257,8 @@
                             <show_in_website>1</show_in_website>
                             <show_in_store>1</show_in_store>
                             <can_be_empty>1</can_be_empty>
-                            <comment><![CDATA[If you enable this Magento confirmation email will be sent for subscribers instead of Mailchimp email.]]></comment>
+                            <comment>
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/etc/system.xml]: This blank lines after the start tag cause failure.

File: app/design/adminhtml/default/default/template/ebizmarts/mailchimp/system/config/form/field/array_dropdown.phtml [hunk 15]
```diff
@@ -118,16 +118,15 @@ $colspan = $colspan > 1 ? 'colspan="' . $colspan . '"' : '';
                 // generate default template data
                 if ('' == templateData) {
                     var d = new Date();
-                    templateData = {
+                    templateData = {;
```
Comment (Santiagoebizmarts) [inline app/design/adminhtml/default/default/template/ebizmarts/mailchimp/system/config/form/field/array_dropdown.phtml]: Why this semicolon?

File: app/design/adminhtml/default/default/template/ebizmarts/mailchimp/system/config/form/field/array_dropdown.phtml [hunk 16]
```diff
@@ -118,16 +118,15 @@ $colspan = $colspan > 1 ? 'colspan="' . $colspan . '"' : '';
...
                     <?php foreach ($this->_columns as $columnName => $column):?>
                     <?php echo $this->quoteEscape($columnName) ?> :
                     '',
                     <?php endforeach;?>
-                        _id
+                        _id;
```
Comment (Santiagoebizmarts) [inline app/design/adminhtml/default/default/template/ebizmarts/mailchimp/system/config/form/field/array_dropdown.phtml]: Why this semicolon?

File: app/design/frontend/base/default/template/ebizmarts/mailchimp/popup/emailcatcher.phtml [hunk 17]
```diff
@@ -59,7 +59,7 @@
             });
             var cookieStored = emailCookieExists();
             if (!cookieStored && $('popupDialog') == undefined) {
-                var popupWidth =
+                var popupWidth =;
```
Comment (Santiagoebizmarts) [inline app/design/frontend/base/default/template/ebizmarts/mailchimp/popup/emailcatcher.phtml]: Why this semicolon?

File: app/design/frontend/base/default/template/ebizmarts/mailchimp/popup/emailcatcher.phtml [hunk 18]
```diff
@@ -70,7 +70,7 @@
                             )
                         ));
                     ?>*window.innerWidth/100;
-                var popupHeight =
+                var popupHeight =;
```
Comment (Santiagoebizmarts) [inline app/design/frontend/base/default/template/ebizmarts/mailchimp/popup/emailcatcher.phtml]: Why this semicolon?

File: app/design/frontend/base/default/template/ebizmarts/mailchimp/popup/emailcatcher.phtml (hunk 1 above)
Comment (Santiagoebizmarts) [reply app/design/frontend/base/default/template/ebizmarts/mailchimp/popup/emailcatcher.phtml]: Is this fully tested?
Can the "new RegExp" instantiation be avoided?

File: lib/Ebizmarts/MailChimp.php [hunk 19]
```diff
@@ -417,4 +423,37 @@ public function call($url, $params, $method = Ebizmarts_MailChimp::GET, $encodeJ
...
+        $this->_curlOptions[$index] = $value;
+    }
+
+    /**
+     * @param array $curlOptions
+     */
+    protected function setCurlOptionsAddOptions($curlOptions = array())
+    {
+        $this->_curlOptions += $curlOptions;
+    }
```
Comment (Santiagoebizmarts) [inline lib/Ebizmarts/MailChimp.php]: What is the purpose of these functions?

File: lib/Mandrill/Mandrill.php [hunk 20]
```diff
@@ -208,4 +227,45 @@ public function log($msg)
...
+        $this->_curlOptions[$index] = $value;
+    }
+
+    /**
+     * @param array $curlOptions
+     */
+    protected function setCurlOptionsAddOptions($curlOptions = array())
+    {
+        $this->_curlOptions += $curlOptions;
+    }
```
Comment (Santiagoebizmarts) [inline lib/Mandrill/Mandrill.php]: What is the purpose of these functions?

File: app/code/community/Ebizmarts/MailChimp/Block/Adminhtml/Mailchimpstores/Edit/Form.php [hunk 21]
```diff
@@ -56,9 +56,8 @@ protected function _prepareForm()
                 $prefix = '';
                 $storeId = $s->getStoreId();
 
-                if(!$helper->ping($storeId)) {
+                if(!$helper->ping($storeId))
```
[the remark this answers is not in the document]
Comment (roberto-ebizmarts) [reply app/code/community/Ebizmarts/MailChimp/Block/Adminhtml/Mailchimpstores/Edit/Form.php]: Solved.

File: app/design/frontend/base/default/template/ebizmarts/mailchimp/popup/emailcatcher.phtml (hunk 1 above)
Comment (roberto-ebizmarts) [reply app/design/frontend/base/default/template/ebizmarts/mailchimp/popup/emailcatcher.phtml]: Confirmed. The current regular expression used for validating email is the same used before, but divided in two parts: user and domain.

File: app/code/community/Ebizmarts/MailChimp/Helper/Data.php [hunk 22]
```diff
@@ -2733,13 +2778,14 @@ public function loadListSubscriber($listId, $email)
...
         $customer = null;
-
         $storeIds = $this->getMagentoStoreIdsByListId($listId);
+
         if (!empty($storeIds)) {
             $customer = Mage::getResourceModel('customer/customer_collection')
                 ->addFieldToFilter('store_id', array('in' => $storeIds))
                 ->addFieldToFilter('email', array('eq' => $email))
-                ->getFirstItem();
+                ->setPageSize(1, 1)->getLastItem();
```
[the remark this answers is not in the document]
Comment (roberto-ebizmarts) [reply app/code/community/Ebizmarts/MailChimp/Helper/Data.php]: Solved.

File: app/code/community/Ebizmarts/MailChimp/Helper/Data.php (hunk 2 above)
Comment (roberto-ebizmarts) [reply app/code/community/Ebizmarts/MailChimp/Helper/Data.php]: Solved

File: app/code/community/Ebizmarts/MailChimp/Helper/Data.php (hunk 3 above)
Comment (roberto-ebizmarts) [reply app/code/community/Ebizmarts/MailChimp/Helper/Data.php]: Solved.

File: app/design/frontend/base/default/template/ebizmarts/mailchimp/popup/emailcatcher.phtml (hunk 17 above)
Comment (roberto-ebizmarts) [reply app/design/frontend/base/default/template/ebizmarts/mailchimp/popup/emailcatcher.phtml]: Solved.

File: app/design/frontend/base/default/template/ebizmarts/mailchimp/popup/emailcatcher.phtml (hunk 18 above)
Comment (roberto-ebizmarts) [reply app/design/frontend/base/default/template/ebizmarts/mailchimp/popup/emailcatcher.phtml]: Solved.

File: lib/Ebizmarts/MailChimp.php (hunk 19 above)
Comment (roberto-ebizmarts) [reply lib/Ebizmarts/MailChimp.php]: They was intended for being used if needed. Now they are removed.

File: lib/Mandrill/Mandrill.php (hunk 20 above)
Comment (roberto-ebizmarts) [reply lib/Mandrill/Mandrill.php]: They was intended for being used if needed. Now they are removed.

File: app/code/community/Ebizmarts/MailChimp/etc/system.xml (hunk 6 above)
Comment (roberto-ebizmarts) [reply app/code/community/Ebizmarts/MailChimp/etc/system.xml]: Solved.

File: app/code/community/Ebizmarts/MailChimp/etc/system.xml (hunk 7 above)
Comment (roberto-ebizmarts) [reply app/code/community/Ebizmarts/MailChimp/etc/system.xml]: Solved.

File: app/code/community/Ebizmarts/MailChimp/etc/system.xml (hunk 8 above)
Comment (roberto-ebizmarts) [reply app/code/community/Ebizmarts/MailChimp/etc/system.xml]: Solved.

File: app/code/community/Ebizmarts/MailChimp/etc/system.xml (hunk 9 above)
Comment (roberto-ebizmarts) [reply app/code/community/Ebizmarts/MailChimp/etc/system.xml]: Solved.

File: app/code/community/Ebizmarts/MailChimp/etc/system.xml (hunk 10 above)
Comment (roberto-ebizmarts) [reply app/code/community/Ebizmarts/MailChimp/etc/system.xml]: Solved.

File: app/code/community/Ebizmarts/MailChimp/etc/system.xml (hunk 11 above)
Comment (roberto-ebizmarts) [reply app/code/community/Ebizmarts/MailChimp/etc/system.xml]: Solved.

File: app/code/community/Ebizmarts/MailChimp/etc/system.xml (hunk 12 above)
Comment (roberto-ebizmarts) [reply app/code/community/Ebizmarts/MailChimp/etc/system.xml]: Solved.

File: app/code/community/Ebizmarts/MailChimp/etc/system.xml (hunk 13 above)
Comment (roberto-ebizmarts) [reply app/code/community/Ebizmarts/MailChimp/etc/system.xml]: Solved.

File: app/code/community/Ebizmarts/MailChimp/etc/system.xml (hunk 14 above)
Comment (roberto-ebizmarts) [reply app/code/community/Ebizmarts/MailChimp/etc/system.xml]: Solved.

File: app/design/adminhtml/default/default/template/ebizmarts/mailchimp/system/config/form/field/array_dropdown.phtml (hunk 15 above)
Comment (roberto-ebizmarts) [reply app/design/adminhtml/default/default/template/ebizmarts/mailchimp/system/config/form/field/array_dropdown.phtml]: Solved.

File: app/design/adminhtml/default/default/template/ebizmarts/mailchimp/system/config/form/field/array_dropdown.phtml (hunk 16 above)
Comment (roberto-ebizmarts) [reply app/design/adminhtml/default/default/template/ebizmarts/mailchimp/system/config/form/field/array_dropdown.phtml]: Solved.

File: app/code/community/Ebizmarts/MailChimp/Helper/Data.php (hunk 4 above)
Comment (roberto-ebizmarts) [reply app/code/community/Ebizmarts/MailChimp/Helper/Data.php]: Solved.

File: app/code/community/Ebizmarts/MailChimp/Model/Api/Subscribers/MailchimpTags.php (hunk 5 above)
Comment (roberto-ebizmarts) [reply app/code/community/Ebizmarts/MailChimp/Model/Api/Subscribers/MailchimpTags.php]: Solved.

**Pull request merged**